File: examples/javaee/src/main/java/grooves/example/javaee/domain/Patient.java
package grooves.example.javaee.domain;

// tag::documented[]
import com.github.rahulsom.grooves.java.Aggregate;

import java.io.Serializable;

@Aggregate // <1>
public class Patient implements Serializable {
    private Long id;
    private String uniqueId;
    // end::documented[]

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getUniqueId() {
        return uniqueId;
    }

    public void setUniqueId(String uniqueId) {
        this.uniqueId = uniqueId;
    }

    @Override
    public String toString() {
        return String.format("Patient{id=%d, uniqueId='%s'}", id, uniqueId);
    }

    public Patient(String uniqueId) {
        this.uniqueId = uniqueId;
    }
    // tag::documented[]
}
// end::documented[]

File: examples/springboot/kotlin/src/main/kotlin/grooves/boot/kotlin/domain/Patient.kt
package grooves.boot.kotlin.domain

// tag::documented[]
import com.github.rahulsom.grooves.api.AggregateType

class Patient(var uniqueId: String? = null) : AggregateType { // <1>
    var id: String? = null // <2>
    override fun toString() = "Patient(uniqueId=$uniqueId, id=$id)"
}
// end::documented[]
